Editor/Scripts/Analytics/MeshsyncAnalytics.cs

```diff
@@ -91,11 +157,84 @@ private struct SyncEventData {
                 entitySyncType = entityTypeStr
             };
 
-            logIfWarning(
-                EditorAnalytics.SendEventWithLimit(
-                    MESHSYNC_SYNC_INSTALLEDEVENTNAME,
-                    eventData,
-                    MESHSYNC_SYNC_VERSION));
+            this.syncEventStack.Push(new TimedEvent<SyncEventData>
+            {
+                LogTime = DateTimeOffset.Now,
+                Data = eventData
+            });
+
+            if (!CoalesceEvents) {
+                var evts = this.syncEventStack.Select(evt => evt.Data).ToArray();
+
+                foreach (var evt in evts) {
+                    logIfWarning(
+                        EditorAnalytics.SendEventWithLimit(
+                            MESHSYNC_SYNC_INSTALLEDEVENTNAME,
+                            evt,
+                            MESHSYNC_SYNC_VERSION));
+                }
+                this.syncEventStack.Clear();
+            }
+            else if ((DateTime.Now - this.lastSync) > MinTimeBetweenSync) {
+
+                // Group entity sync events by type
+                var syncEntityTypeLookup = this.syncEventStack
```

Comment:
Isn't it simpler to check if we have already added the `TimedEvent` to the `combined list`, rather than grouping and sorting them again ? 
Something like this (untested):

```
foreach(var e in syncEventList) {
   float second = ToSecond(evt.LogTime);
    if (isEntity(e)) {
        if (entityAdded.Contains(second))
            continue;
        else 
            entityAdded.Add(second):
    }
    if (isAsset(e)) {
        if (assetAdded.Contains(second))
            continue;
        else 
            assetAdded.Add(second):
    }

    combined.add(e);
}
```

Reply:
@mloccy What do you think of the above idea ?

Reply:
@mloccy One comment left above. Please let us know what you think to proceed with this PR.

Reply:
Hi Sindharta,

Apologies I've been a bit distracted with other duties this week. 

From what I understand it's not exactly the same thing, from reading this code it seemed that this would limit us to one entity / asset sync event per second, rather than one asset sync kind per second. 

The issue we're trying to solve is eliminating duplicates that add no value, a series like 

[13:47:53:0022] Transform
[13:47:53:0044] Transform
[13:47:53:0132] Transform
[13:47:53:0285] Transform ... 
for example

Correct me if I'm wrong but it seems the code that you're suggesting would limit any sync to once per second, as opposed to once per kind per second:

[13:47:53:020] Mesh
[13:47:53:070] Transform
[13:47:53:130] Transform
[13:47:53:180] Transform
[13:47:53:230] Transform
[13:47:53:280] Transform
[13:47:53:330] Transform

The code currently in the patch would collapse this to:

[13:47:53:020] Mesh
[13:47:53:070] Transform

Whereas the code above seems to collapse this to:
[13:47:53:020] Mesh

If that is the case we then lose that user behaviour information about how users are using MeshSync, does that make sense?

Reply:
@mloccy 
I see, but doesn't the above code also apply to the purpose you are suggesting above with a small change (a dictionary of a dictionary, rather than a dictionary) ?

Something like:
```
        foreach (var e in syncEventList) {
            float second = ToSecond(evt.LogTime);
            if (isEntity(e)) {
                if (entityAdded[kind].Contains(second)) continue;
                else entityAdded[kind].Add(second):
            }

            if (isAsset(e)) {
                if (assetAdded[kind].Contains(second)) continue;
                else assetAdded[kind].Add(second):
            }

            combined.add(e);
        }  
```